tests/old_functional/test_direct_transfer.py

```diff
@@ -838,6 +838,155 @@ def test_sessions(self):
                     matches = list(filter(expression.match, records))
                     assert len(matches) == x + 1
 
+    def test_pause_resume_session(self):
+        """
+        Test the session pause and resume system.
+        The Session final status should be COMPLETED.
+        """
+
+        # There is no upload, right now
+        assert not list(self.engine_1.dao.get_dt_uploads())
+        expression = re.compile(
+            r"<LogRecord: nxdrive\.notification, .*, .*, .*, "
+            r"\"Sending Notification\(level='info' title='Direct Transfer'"
+            r" uid='DIRECT_TRANSFER_SESSION_END.*' unique=False\)\">"
+        )
+
+        def callback(*_):
+            """This will mimic what is done in SessionItem.qml."""
+            # Ensure we have 1 ongoing upload
+            dao = self.engine_1.dao
+            uploads = list(dao.get_dt_uploads())
+            assert uploads
+            upload = uploads[0]
+            assert upload.status == TransferStatus.ONGOING
+
+            # Verify the session status
+            sessions = dao.get_active_sessions_raw()
+            assert len(sessions) == 1
+            session = sessions[0]
+            assert session["total"] == 2
+            assert session["status"] == TransferStatus.ONGOING
+
+            # Pause the session
+            dao.pause_session(session["uid"])
+
+            session = self.engine_1.dao.get_session(session["uid"])
+            print(session)
+            uploads = list(dao.get_dt_uploads())
+            assert uploads
+            upload = uploads[0]
+            assert upload.status == TransferStatus.PAUSED
+
+        created = []
+        root_folder = self.tmpdir / str(uuid4())[:6]
+        root_folder.mkdir()
+        created.append(root_folder)
+
+        sub_file = root_folder / f"file_{str(uuid4())[:4]}"
+        sub_file.write_text("Some content." * 1024 * 1024 * 2, encoding="utf8")
+        created.append(sub_file)
+
+        with patch.object(self.engine_1.remote, "upload_callback", new=callback):
+            with ensure_no_exception():
+                self.direct_transfer(root_folder)
+                self.wait_sync()
+
+                session = self.engine_1.dao.get_session(1)
+                assert session
+                assert session.status == TransferStatus.PAUSED
+
+        self.engine_1.resume_session(1)
+        with self._caplog.at_level(logging.INFO):
+            self.wait_sync(wait_for_async=True)
+
+            sessions = self.engine_1.dao.get_completed_sessions_raw()
+            assert sessions
+            assert len(sessions) == 1
+            session = sessions[0]
+            assert session["status"] == TransferStatus.DONE
+
+            # A new Notification logs should appear at each iteration
+            records = [str(log) for log in self._caplog.records]
```

Comment:
```suggestion
            records = map(str, self._caplog.records)
```

(same remarks for all new tests)